uvicorn/server.py

```diff
@@ -126,6 +135,7 @@ def _share_socket(sock: socket.SocketType) -> socket.SocketType:
             uds_perms = 0o666
             if os.path.exists(config.uds):
                 uds_perms = os.stat(config.uds).st_mode
+            assert hasattr(asyncio, "start_unix_server")  # mypy
```

Comment:
```suggestion
            assert sys.platform != "win32"  # mypy
```

Reply:
ah you can't use `assert sys.platform` you need `if sys.platform != "win32"`

Reply:
https://github.com/python/mypy/issues/9025

Reply:
```python
        elif config.uds is not None:
            # Create a socket using UNIX domain socket.
            uds_perms = 0o666
            if os.path.exists(config.uds):
                uds_perms = os.stat(config.uds).st_mode
            if sys.platform != "win32":
                server = await asyncio.start_unix_server(
                    handler, path=config.uds, ssl=config.ssl, backlog=config.backlog
                )
                os.chmod(config.uds, uds_perms)
                assert server.sockets is not None  # mypy
                listeners = server.sockets
                self.servers = [server]
 ```

works for me when I run with `mypy --platform=win32`

Reply:
This might be a bit neater?

```diff
From 2b977657b86649222550daf1957493dd6ca40fc5 Mon Sep 17 00:00:00 2001
From: Thomas Grainger <tagrain@gmail.com>
Date: Sat, 26 Jun 2021 14:56:44 +0100
Subject: [PATCH] conditional typecheck start_unix_server

---
 scripts/check     |  2 +-
 uvicorn/server.py | 12 +++++++++---
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/scripts/check b/scripts/check
index e319add..eed005d 100755
--- a/scripts/check
+++ b/scripts/check
@@ -10,6 +10,6 @@ set -x
 
 ${PREFIX}black --check --diff --target-version=py36 $SOURCE_FILES
 ${PREFIX}flake8 $SOURCE_FILES
-${PREFIX}mypy --show-error-codes
+${PREFIX}mypy --platform=win32 --show-error-codes
 ${PREFIX}isort --check --diff --project=uvicorn $SOURCE_FILES
 ${PREFIX}python -m tools.cli_usage --check
diff --git a/uvicorn/server.py b/uvicorn/server.py
index 9ac565b..8a5993c 100644
--- a/uvicorn/server.py
+++ b/uvicorn/server.py
@@ -9,7 +9,7 @@ import threading
 import time
 from email.utils import formatdate
 from types import FrameType
-from typing import List, Optional, Set, Tuple, Union
+from typing import Any, List, Optional, Set, Tuple, Union
 
 import click
 
@@ -30,6 +30,13 @@ logger = logging.getLogger("uvicorn.error")
 
 Protocols = Union[H11Protocol, HttpToolsProtocol, WSProtocol, WebSocketProtocol]
 
+if sys.platform != "win32":
+    from asyncio import start_unix_server as _start_unix_server
+else:
+
+    async def _start_unix_server(*args: object, **kwargs: object) -> Any:
+        raise NotImplementedError("Cannot start a unix server on win32")
+
 
 class ServerState:
     """
@@ -135,8 +142,7 @@ class Server:
             uds_perms = 0o666
             if os.path.exists(config.uds):
                 uds_perms = os.stat(config.uds).st_mode
-            assert hasattr(asyncio, "start_unix_server")  # mypy
-            server = await asyncio.start_unix_server(
+            server = await _start_unix_server(
                 handler, path=config.uds, ssl=config.ssl, backlog=config.backlog
             )
             os.chmod(config.uds, uds_perms)
-- 
2.32.0
```

Reply:
Thank you, @graingert ! Precise as always 😎

I do prefer your solution than using assert (it's not like I have an alternative 😅). But I wonder, isn't it too much to please `mypy`? Does my question even makes sense? 🤔

Anyway, I'm going to update this branch with the suggestion as soon as I get on my computer.

Reply:
https://bugs.python.org/issue33408